mindsdb/integrations/handlers/mongodb_handler/mongodb_handler.py

```diff
@@ -62,11 +62,6 @@ def connect(self):
         if re.match(r'/?.*tls=false', self.host.lower()):
             kwargs['tls'] = False
 
-        if re.match(r'.*.mongodb.net', self.host.lower()) and kwargs.get('tls', None) is None:
```

Comment:
@ZoranPandovski From what I have understood, this block of code was added to ensure secure connections to MongoDB Atlas. Is it OK to remove it?

I think we should also be able to solve the CodeQL alert by changing the regular expression as follows?
```
if re.match(r'.*\.mongodb.net', self.host.lower()) and kwargs.get('tls', None) is None:
    kwargs['tlsCAFile'] = certifi.where()
    if kwargs.get('tls', None) is None:
        kwargs['tls'] = True
```

By escaping the dot before `mongodb.net`, we can ensure that this dot matches a literal dot. This will not match other hosts as the CodeQL alert suggests. This is only necessary if keeping this block of code is required.

Reply:
Yes, this is not needed anymore. I've tested Atlas connections, and it works; you can check the screenshot above. Mongo Atlas provides certificate to establish the connection. You can check [pymongo docs](https://pymongo.readthedocs.io/en/stable/examples/tls.html#tls-ssl-and-pymongo).

Reply:
Understood. Thanks, Zoran.